test/PandoraInterface.cxx

```diff
@@ -336,14 +368,464 @@ void ProcessEvents(const Parameters &parameters, const Pandora *const pPrimaryPa
+    if (hitLength < std::numeric_limits<float>::epsilon())
+    {
+        //std::cout << "Cannot have zero track length" << std::endl;
+        return currentVoxelList;
+    }
+
+    // Define ray trajectory (this checks dirMag >= epsilon)
+    const CartesianVector dirNorm = dir.GetUnitVector();
+    LArRay ray(start, dirNorm);
+
+    // We need to shuffle along the hit segment path and create voxels as we go.
+    // There are 4 cases for the start and end points inside the voxelisation region.
+    // Case 1: start & stop are both inside the voxelisation boundary
+    // Case 2: start & stop are both outside, but path direction intersects boundary
+    // Case 3: start is inside boundary, stop = intersection at region boundary
+    // Case 4: end is inside boundary, start = intersection at region boundary
+
+    double t0(0.0), t1(0.0);
+    pandora::CartesianVector point1(0.0, 0.0, 0.0), point2(0.0, 0.0, 0.0);
+
+    // Check if the start and end points are inside the voxelisation region
+    const bool inStart = grid.inside(start);
+    const bool inStop = grid.inside(stop);
+
+    if (inStart && inStop)
+    {
+        //std::cout << "Case 1: Start and end points are inside boundary" << std::endl;
+        point1 = start;
+        point2 = stop;
+    }
+    else if (!inStart && !inStop)
+    {
+        //std::cout << "Case 2: Start and end points are outside boundary" << std::endl;
+        const bool ok = grid.intersect(ray, t0, t1);
+        if (ok)
+        {
+            point1 = ray.getPoint(t0);
+            point2 = ray.getPoint(t1);
+            //std::cout << "Boundary points: " << point1 << " and " << point2 << std::endl;
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary limits." << std::endl;
+            return currentVoxelList;
+        }
+    }
+    else if (inStart && !inStop)
+    {
+        //std::cout << "Case 3: Start inside boundary" << std::endl;
+        point1 = start;
+        const bool ok = grid.intersect(ray, t0, t1);
+        if (ok)
+        {
+            point2 = ray.getPoint(t1);
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary" << std::endl;
+            return currentVoxelList;
+        }
+    }
+    else if (!inStart && inStop)
+    {
+        //std::cout << "Case 4: End inside boundary" << std::endl;
+        point2 = stop;
+        const bool ok = grid.intersect(ray, t0, t1);
+        if (ok)
+        {
+            point1 = ray.getPoint(t0);
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary" << std::endl;
+            return currentVoxelList;
+        }
+    }
+
+    // Now create voxels between point1 and point2.
+    // Ray direction will be the same, but update starting point
+    ray.updateOrigin(point1);
+
+    // Small position shift to ensure we move along the ray path
+    const double epsilon(1e-3);
```

Comment:
How does the value of this impact the efficiency? Any concerns at all about efficiency?

Reply:
Don't think we have tried to find an optimal value for this. We just need a small path length change (but not too small) so that we continue along the hit segment "trajectory" to find the next (different) voxel bin.

Reply:
This path shift has been added as the constant value `m_voxelPathShift` in `Parameters`. Allows us the possibility to include this as an optional runtime parameter within `ParseCommandLine` if we discover efficiency issues related to this.